external-resolver/src/main/scala/org/genivi/sota/resolver/devices/DeviceDirectives.scala

```diff
@@ -40,7 +43,16 @@ class DeviceDirectives(namespaceExtractor: Directive1[Namespace],
 
   import Directives._
 
-  val extractUuid: Directive1[Uuid] = refined[Uuid.Valid](Slash ~ Segment).map(Uuid(_))
+  private[this] def failNamespaceRejection(msg: String): Rejection = AuthorizationFailedRejection
+
+  def extractDeviceUuid(ns: Namespace): Directive1[Uuid] = extractUuid.flatMap { deviceId =>
+    import scala.util.{Success, Failure}
+    val f = deviceRegistry.fetchDevice(ns, deviceId)
+    onComplete(f) flatMap {
+      case Success(device) => provide(deviceId)
+      case Failure(t) => reject(failNamespaceRejection("Cannot validate namespace"))
```

Comment:
Also we should check if `t` is the right error otherwise we will 401 when db is down